internal/parser.go

```diff
@@ -655,8 +655,13 @@ func beforeHTMLIM(p *parser) bool {
 			return true
 		}
 		if isComponent(p.tok.Data) {
+			p.parseImpliedToken(StartTagToken, a.Html, a.Html.String())
+			p.parseImpliedToken(StartTagToken, a.Head, a.Head.String())
 			p.addElement()
 			p.im = inBodyIM
+			if p.hasSelfClosingToken {
+				p.oe.pop()
```

Comment:
This is what fixes self-closing components having siblings pushed into children.